internal/core/workers/runtime_watcher_worker/runtime_watcher_worker.go

```diff
@@ -71,20 +72,27 @@ func (w *runtimeWatcherWorker) Start(ctx context.Context) error {
 
 	resultChan := watcher.ResultChan()
 
-	// TODO(guilhermocc): We need to deal better with this events chan processing by using https://github.com/kubernetes/client-go/blob/master/util/workqueue/doc.go
+	workerWaitGroup := &sync.WaitGroup{}
+
 	for i := 0; i < 200; i++ {
+		workerWaitGroup.Add(1)
 		go func(goroutineNumber int) {
+			defer workerWaitGroup.Done()
 			goroutineLogger := w.logger.With(zap.Int("goroutine", goroutineNumber))
 			goroutineLogger.Info("Starting event processing goroutine")
 			for {
 				select {
 				case event, ok := <-resultChan:
 					if !ok {
+						w.logger.Warn("resultChan closed, finishing worker goroutine", zap.Error(err))
 						return
 					}
-					err := w.processEvent(w.ctx, event)
+					err = w.processEvent(w.ctx, event)
 					if err != nil {
 						w.logger.Warn("failed to process event", zap.Error(err))
+						reportEventProcessingStatus(event, false)
+					} else {
+						reportEventProcessingStatus(event, true)
 					}
```

Comment:
```suggestion
					}
					
					reportEventProcessingStatus(event, true)
```

Reply:
we can not do this because the function is not returning I fail branch